ibis/bigquery/client.py

```diff
@@ -286,38 +250,53 @@ def bq_param_date(param, value):
     return bq.ScalarQueryParameter(param.get_name(), 'DATE', value)
 
 
+class BigQueryTable(ops.DatabaseTable):
+    pass
+
+
+def rename_partitioned_column(table_expr, bq_table):
+    partition_info = bq_table._properties.get('timePartitioning', None)
+
+    # If we don't have any partiton information, the table isn't partitioned
+    if partition_info is None:
+        return table_expr
+
+    # If we have a partition, but no "field" field in the table properties,
+    # then use NATIVE_PARTITION_COL as the default
+    partition_field = partition_info.get('field', NATIVE_PARTITION_COL)
+
+    # The partition field must be in table_expr columns
+    assert partition_field in table_expr.columns
+
+    # User configured partition column name default
+    col = ibis.options.bigquery.partition_col
+    if col in table_expr.columns:
+        return table_expr
+    return table_expr.relabel({partition_field: col})
+
+
 class BigQueryClient(SQLClient):
 
     sync_query = BigQueryQuery
     database_class = BigQueryDatabase
-    proxy_class = BigQueryAPIProxy
+    table_class = BigQueryTable
```

Comment:
The functionality to make this work isn't yet implemented. I've got a PR in the works to do this.